modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/dto/converter/DTOConverterRegistryImpl.java

```diff
@@ -35,33 +44,143 @@ public class DTOConverterRegistryImpl implements DTOConverterRegistry {
 
 	@Override
 	public Set<String> getDTOClassNames() {
-		return _serviceTrackerMap.keySet();
+		return _dtoConvertersMap.keySet();
 	}
 
 	@Override
 	public DTOConverter<?, ?> getDTOConverter(String dtoClassName) {
-		return _serviceTrackerMap.getService(dtoClassName);
+		List<DTOConverter<?, ?>> dtoConverters = _dtoConvertersMap.get(
+			dtoClassName);
+
+		if (ListUtil.isEmpty(dtoConverters)) {
+			return null;
+		}
+
+		return dtoConverters.get(0);
+	}
+
+	@Override
+	public DTOConverter<?, ?> getDTOConverter(
+		String applicationName, String dtoClassName, String version) {
+
+		List<DTOConverter<?, ?>> dtoConverters = _dtoConvertersMap.get(
+			dtoClassName);
+
+		if (ListUtil.isEmpty(dtoConverters)) {
+			return null;
+		}
+
+		for (DTOConverter<?, ?> dtoConverter : dtoConverters) {
```

Comment:
Instead of looping, we could create a separate serviceTrackerMap that emits a composite key (model+app+version) and then we can just to a single get. But I think this is good enough for now.

Reply:
I agree it is a nice option! I will change it